### PR TITLE
On generated doc error, don't try to download the file (return 500 instead)

### PR DESCRIPTION
**Pull request by jonnywyatt**

### CODE REVIEW

File: server/controllers/documents/generated/createGeneratedDocument.ts [hunk 1]
```diff
@@ -31,9 +31,9 @@ export const createGeneratedDocument = async (req: Request, res: Response, next:
       const { documentId } = await generateRecallDocument(recallId, { category: cat, fileName }, token)
       res.locals.documentId = documentId
     }
+    next()
   } catch (err) {
     logger.error(err)
-  } finally {
-    next()
+    res.sendStatus(500)
```
Comment (philsegal) [inline server/controllers/documents/generated/createGeneratedDocument.ts:37]: how is this displayed to end-user? Presumably we'd like them to try again?

File: integration_tests/integration/assess-recall.spec.js [hunk 2]
```diff
@@ -76,7 +76,7 @@ context('Assess a recall', () => {
     cy.getText('missing-OASYS_RISK_ASSESSMENT').should('equal', 'Missing')
   })
 
-  it('can assess a recall', () => {
+  it.only('can assess a recall', () => {
```
[the remark this answers is not in the document]
Comment (jonnywyatt) [reply integration_tests/integration/assess-recall.spec.js:79]: oops!

[the remark this answers is not in the document]
Comment (jonnywyatt) [reply integration_tests/integration/assess-recall.spec.js]: Removed

File: server/controllers/documents/generated/createGeneratedDocument.ts (hunk 1 above)
Comment (jonnywyatt) [reply server/controllers/documents/generated/createGeneratedDocument.ts:37]: I looked into this but right now it just doesn't download it. We could certainly improve the UX, lets discuss with product. Maybe worth a spike. It's not straightforward as far as I can see, to issue a redirect on the download request and have it actioned by the browser.

Comment (philsegal) [reply server/controllers/documents/generated/createGeneratedDocument.ts:37]: Cool, makes sense. I guess an error on the page would be ideal, is the 'download action' also doing the generate call as well? so its not that we can return the generate error to the UI before we 'open' the download.

**Review by philsegal — APPROVED** [review]

**Pull request merged**